torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -531,44 +529,46 @@ class CppMicroGemmAMX(CppMicroGemm):
-        auto b_int8 = at::vec::Vectorized<int8_t>::loadu(src, static_cast<int64_t>(32));
-        auto b_bf16 = at::vec::convert<{{input_t}}>(b_int8);
-        b_bf16.store(dst);
-    };
-
-    auto load_B_tile = [&]({{input2_t}}* B_ptr, int idx, int num_b_rows) {
-        {{input_t}}* base_addr = dequantized_B_buf + idx;
-        {{kernel.unroll_pragma(8)}}
-        for (int i = 0; i < num_b_rows; i++) {
-            load_B_row(
-                B_ptr + i * b_tile_ptr_stride,
-                base_addr + i * 32
-            );
-        }
-    };
-    auto load_dequantized_B = [&](int n) {
+    auto load_dequantized_B = [&](int base_idx) {
         // Load a tile of B & cache it in L1D.
-        {{kernel.unroll_pragma(4)}}
-        for (int k = 0; k < K; k += {{block_k}}) {
-            int num_b_rows = (k < last_k_offset) ? 16 : tail_k_size;
-            {{kernel.unroll_pragma(2)}}
-            for (int tile_col = 0; tile_col <= 1; tile_col++) {
-                load_B_tile(
-                    const_cast<{{input2_t}}*>(B) + n + k * ldb + tile_col * {{16 * vnni_size}},
-                    (k / {{block_k // 2}} + tile_col) * num_elements_per_b_tile,
-                    num_b_rows
-                );
-            }
+        {{input2_t}}* base_addr = const_cast<{{input2_t}}*>(B) + base_idx;
+        for (int idx_dq = 0, idx_q = 0; idx_dq < buf_size; idx_q += ldb, idx_dq += block_n) {
+        {%- if block_n in [16, 32] %}
+            auto b_int8 = at::vec::Vectorized<int8_t>::loadu(
+                base_addr + idx_q,
+                static_cast<int64_t>(block_n)
+            );
+            auto b_bf16 = at::vec::convert<{{input_t}}>(b_int8);
+            b_bf16.store(
+                dequantized_B_buf + idx_dq,
+                static_cast<int64_t>(block_n)
+            );
+        {%- elif block_n == 48 %}
+            // first 32 elements
+            auto b_int8_32 = at::vec::Vectorized<int8_t>::loadu(
+                base_addr + idx_q,
+                static_cast<int64_t>(32)
+            );
+            auto b_bf16_32 = at::vec::convert<{{input_t}}>(b_int8_32);
+            b_bf16_32.store(dequantized_B_buf + idx_dq);
+            // next 16 elements
+            auto b_int8_16 = at::vec::Vectorized<int8_t>::loadu(
+                base_addr + idx_q + 32,
+                static_cast<int64_t>(16)
+            );
+            auto b_bf16_16 = at::vec::convert<{{input_t}}>(b_int8_16);
+            b_bf16_16.store(
+                dequantized_B_buf + idx_dq + 32,
+                static_cast<int64_t>(16)
+            );
+        {%- endif %}
         }
     };
 {%- endif %}
     // TODO(jgong5): loop unroll for M and N
-    for (int64_t n = 0; n < N; n += {{block_n}}) {
+    for (int64_t n = 0; n < N; n += block_n) {
 {%- if use_cached_dequantized_B %}
-        // Dequantize K * 32 int8 B elements into BF16
+        // Dequantize K * block_n int8 B elements into BF16
```

Comment:
I guess for all these comments you can directly use `{{block_n}}` instead of `block_n`.